spec/factories/knights.rb

```diff
@@ -0,0 +1,4 @@
+FactoryGirl.define do
```

Comment:
Is this being used at all?

Seems like no as you're creating Pieces in the test rather than here with traits--